leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -107,7 +107,7 @@ private AwaitingLeadershipProxy(
                 "Unable to create an AwaitingLeadershipProxy with no supplier");
         this.delegateSupplier = delegateSupplier;
         this.leaderElectionService = leaderElectionService;
-        this.executor = PTExecutors.newSingleThreadExecutor(PTExecutors.newNamedThreadFactory(true));
+        this.executor = PTExecutors.newSingleThreadExecutor();
```

Comment:
Feels like this should have a name?

Reply:
So this will loose daemon threading? @jeremyk-91

Reply:
This uses the same logic as before, discovering a name based on the call site.

Reply:
This used daemon threads before, and the default is using daemon threads. This one hasn't changed, only the psl threads are changing to daemons.